packages/core/src/asciidoctor-core-api.js

```diff
@@ -2129,3 +2129,111 @@ Timings.prototype.printReport = function (to, subject) {
+ *
+ * @memberof SyntaxHighlighter
+ * @param {string} name - The {string} name of the syntax highlighter to create.
+ * @param {string} backend - The {string} name of the backend for which this syntax highlighter is being used (default: 'html5').
+ * @param {Object} opts - a JSON of options providing information about the context in which this syntax highlighter is used:
+ * - doc: The Document for which this syntax highlighter was created.
+ * @returns a {SyntaxHighlighter} instance for the specified name.
+ */
+SyntaxHighlighter.create = function (name, backend, opts) {
+  if (typeof backend === 'undefined') backend = 'html5'
+  if (typeof opts === 'undefined') opts = {}
+  return SyntaxHighlighter.$create(name, backend, opts)
+}
+
+/**
+ * @namespace
+ */
+var SyntaxHighlighterBase = Opal.const_get_qualified(SyntaxHighlighter, 'Base', true)
+
+// Alias
+Opal.Asciidoctor.SyntaxHighlighterBase = SyntaxHighlighterBase
+
+/**
+ * Creates a new {SyntaxHighlighter} instance.
+ *
+ * @memberof SyntaxHighlighterBase
+ * @param {string} name - The {string} name of the syntax highlighter to create.
+ * @param {Function} block - A {Function} with the {SyntaxHighlighter} instance passed as `this`
+ * @returns a {SyntaxHighlighter} instance for the specified name.
+ */
+SyntaxHighlighterBase.create = function (name, block) {
+  var instance = Opal.klass(Opal.Object, SyntaxHighlighterBase, name, function () {}).$new(name)
+  instance.registerFor(name)
+  if (typeof block === 'function') {
+    block.apply(instance, [])
+  }
+  return instance
+}
+
+/**
+ * Statically register the current class in the registry for the specified names.
+ *
+ * @memberof SyntaxHighlighterBase
+ * @param {string|Array} names - A {string} name or an {Array} of {string} names
+ */
+SyntaxHighlighterBase.prototype.registerFor = function (names) {
+  SyntaxHighlighter.register(this, names)
+}
+
+/**
+ * Format the highlighted source for inclusion in an HTML document.
+ *
+ * @memberof SyntaxHighlighterBase
+ * @param {Function} block - A {Function} that returns the highlighted source {string} wrapped in preformatted tags (e.g., pre and code).
+ * The {Function} takes 3 arguments:
+ * node: The source Block being processed.
+ * lang: The source language String for this Block (e.g., ruby).
+ * opts: A Hash of options that control syntax highlighting:
+ * - nowrap: A Boolean that indicates whether wrapping should be disabled (optional).
+ */
+SyntaxHighlighterBase.prototype.format = function (block) {
+  Opal.def(this, '$format', block)
+}
+
+/**
+ * Indicates whether highlighting is handled by this syntax highlighter or by the client.
+ *
+ * @memberof SyntaxHighlighterBase
+ * @param {Function} block - A {Function} that returns a {Boolean} indicating whether the highlight method should be used to handle the specialchars substitution.
+ */
+SyntaxHighlighterBase.prototype.highlighted = function (block) {
```

Comment:
I'm not sure about the naming, `isHighlight` feels weird, what do you think ?

Reply:
Maybe `isHighlighted` ?

Reply:
The intended meaning of the name here is "shouldHighlight" or "handlesHighlighting". The premise is whether this adapter handles highlighting, or whether it just prepares highlighting to be performed by the viewer.